specs/ClientCertificate.md

```diff
@@ -0,0 +1,515 @@
+programmatically query the certificates and select a certificate from the list to respond to the server.
+
+In this document we describe the API. We'd appreciate your feedback.
+
+
+# Description
+We propose adding `ClientCertificateRequested` API that allows you to query the mutually trusted CA
+certificates, and select the certificate from the list to respond to the server.
+
+When the event is raised, WebView2 will pass a `CoreWebView2ClientCertificateRequestedEventArgs`, which lets
+you to view the server information that requested client certificate authentication, list of certificates filtered
+from issuers that the server trusts, inspect the certificate metadata and serveral options for responding to client certificate request.
+
+- You can select a certificate from the list to respond to the server.
+- You can choose to respond to the server without a certificate.
+- You can choose to display default client certificate selection dialog prompt to let user to respond to the server.
+- You can cancel the request.
+
+# Examples
+The following code snippets demonstrate how the ClientCertificateRequested event can be used:
+## Win32 C++
+```cpp
+// Turn on or off client certificate selection dialog by adding or removing a
+// ClientCertificateRequested handler which intercepts when server requests client
+// certificate for authentication.
+// This example hides the default client certificate dialog and allows you to set
+// the client certificate programmatically.
+void SettingsComponent::SetClientCertificateRequested(bool raiseClientCertificate)
+{
+    if (raiseClientCertificate != m_raiseClientCertificate)
+    {
+        m_raiseClientCertificate = raiseClientCertificate;
+        //! [ClientCertificateRequested1]
+        if (m_raiseClientCertificate)
+        {
+            CHECK_FAILURE(m_webView->add_ClientCertificateRequested(
+                Callback<ICoreWebView2ClientCertificateRequestedEventHandler>(
+                    [this](
+                        ICoreWebView2* sender,
+                        ICoreWebView2ClientCertificateRequestedEventArgs* args) {
+
+                        ICoreWebView2ClientCertificateList* certificateList;
+                        CHECK_FAILURE(args->get_MutuallyTrustedCertificates(&certificateList));
+
+                        UINT cert_lst_size;
+                        CHECK_FAILURE(certificateList->get_Count(&cert_lst_size));
+
+                        ICoreWebView2ClientCertificate* certificate = nullptr;
+
+                        if (cert_lst_size > 0)
+                        {
+                            for (UINT i = 0; i < cert_lst_size; i++)
+                            {
+                                CHECK_FAILURE(
+                                    certificateList->GetValueAtIndex(i, &certificate));
+                            }
+                        }
+
+                        if (certificate)
```

Comment:
Browser doesn't show a dialog to the user when mutually trusted certificates are zero and respond to the server without a client certificate and hence we don't raise an event in WebView2. The count will be always greater than 0 with in the event. To avoid confusion, I removed the check for certificate list count and responding to the server with the selected certificate (last in the list) in the example.

Also, added a comment at line#48 and #298